# <center> ESTATÍSTICA APLICADA </center>

# <center>  Projeto 1 </center>

# <center>  José Silvestre da Silva Galvão </center>

### <center> Tarefa 1 </center>

Sim, é possível acessar os arquivos compactados diretamente do site sem precisar baixar.

In [8]:
import pandas as pd
import numpy as np


Para facilitar o meu trabalho coloquei os links em uma váriavel.

In [9]:
arq2018 = "https://arquivos.prf.gov.br/arquivos/index.php/s/EF4uPKCihT0ouXd/download"
arq2019 = "https://arquivos.prf.gov.br/arquivos/index.php/s/sdvJndbl5wLyh3J/download"
arq2020 = "https://arquivos.prf.gov.br/arquivos/index.php/s/hXimwPNR9lyqdMS/download"

Fiz a descompactação do arquivo diretamente do site usando o compression zip. E depois gerei o dataframe.

In [10]:
df2018 = pd.read_csv(arq2018, compression = 'zip', encoding='iso8859-1', sep = ';')
df2018.head()

id     pesid data_inversa     dia_semana   horario  uf     br     km  \
0  99973.0  216523.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1  99973.0  216524.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
2  99973.0  216532.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
3  99973.0  216527.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
4  99973.0  216530.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   

  municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0   RESENDE             Sim  ...  Masculino       0             0   
1   RESENDE             Sim  ...   Feminino       0             0   
2   RESENDE             Sim  ...  Masculino       1             0   
3   RESENDE             Sim  ...   Feminino       0             0   
4   RESENDE             Sim  ...  Masculino       0             0   

  feridos_graves mortos   latitude  longitude regional delegacia       uop  
0              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
1              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
2              0      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
3              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
4              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  

[5 rows x 37 columns]

In [11]:
df2019 = pd.read_csv(arq2019, compression = 'zip', encoding ='iso8859-1', sep = ';')
df2019.head()

id     pesid data_inversa   dia_semana   horario  uf     br     km  \
0  182210.0  402103.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
1  182210.0  402106.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
2  182210.0  402104.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
3  182210.0  402102.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
4  182211.0  402126.0   2019-01-01  terça-feira  01:30:00  PR  373.0  177,3   

      municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0     GUARULHOS             Sim  ...  Masculino       0             1   
1     GUARULHOS             Sim  ...  Masculino       0             1   
2     GUARULHOS             Sim  ...   Feminino       0             1   
3     GUARULHOS             Sim  ...  Masculino       0             1   
4  PONTA GROSSA             Sim  ...  Masculino       0             1   

  feridos_graves mortos      latitude     longitude regional delegacia  \
0              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
1              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
2              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
3              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
4              0      0  -25,05533957  -50,22776753    SR-PR    DEL7/3   

        uop  
0  UOP01/SP  
1  UOP01/SP  
2  UOP01/SP  
3  UOP01/SP  
4  UOP01/PR  

[5 rows x 37 columns]

In [12]:
df2020 = pd.read_csv(arq2020, compression = 'zip', encoding = 'utf_8_sig', sep = ';')
df20 = df2020.replace({'terça':'terça-feira','quarta':'quarta-feira','quinta':'quinta-feira','sexta':'sexta-feira'})
df20.head()

id     pesid data_inversa    dia_semana   horario  uf     br     km  \
0  260031.0  578987.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1  260031.0  578986.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
2  260031.0  578988.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
3  260031.0  578991.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
4  260031.0  578475.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   

  municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0    GURUPI             Sim  ...  Masculino       0             0   
1    GURUPI             Sim  ...   Feminino       0             0   
2    GURUPI             Sim  ...  Masculino       0             0   
3    GURUPI             Sim  ...   Feminino       0             0   
4    GURUPI             Sim  ...   Feminino       0             0   

  feridos_graves mortos      latitude     longitude regional delegacia  uop  
0              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
1              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
2              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
3              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
4              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  

[5 rows x 37 columns]

### <center> Tarefa 2 </center>

#### A)

#### Para saber a quantidade de acidentes que tiveram, usei o len().

In [13]:
len(df2018)

324809

Em 2018 aconteceram 324809 acidentes

In [14]:
len(df2019)

331666

No ano de 2019 aconteceram 331666 acidentes

In [15]:
len(df20)

394482

E no ano de 2020 aconteceram 394482 acidentes.

#### B)

#### Usei a função shape() para retornar o numero de linhas e o numero de colunas do dataframe. Onde o numero de variaveis são o número de colunas.

In [16]:
df2018.shape

(324809, 37)

In [17]:
df2019.shape

(331666, 37)

In [18]:
df20.shape

(394482, 37)

Nos anos de 2018, 2019 e 2020, existem no total 37 variaveis em cada um.

#### C)

In [19]:
df2018['municipio'].value_counts()

CURITIBA             3945
BRASILIA             3784
SAO JOSE             3398
GUARULHOS            3052
BETIM                2741
                     ... 
SAO FERNANDO            1
SAO BENTINHO            1
MOGI DAS CRUZES         1
CUNHA                   1
CAJUEIRO DA PRAIA       1
Name: municipio, Length: 1782, dtype: int64

Em 2018 os municipios com mais acidentes foram, Curitiba com 3945 acidentes, Brasilia com 3784, São José com 3398, Guarulhos 3052 e Betim com 2741 acidentes. 

In [20]:
df2019['municipio'].value_counts()

BRASILIA            5274
CURITIBA            4622
SAO JOSE            3693
GUARULHOS           3477
FORTALEZA           2523
                    ... 
PEDRO LEOPOLDO         1
PEDRAS DE FOGO         1
PRADO                  1
FRANCISCO MACEDO       1
PLACAS                 1
Name: municipio, Length: 1767, dtype: int64

Em 2019 os municipios com mais acidentes foram, Brasilia com 5274 acidentes, Curitiba com 4622, São José com 3693, Guarulhos 3477 e Fortaleza com 2523 acidentes. 

In [21]:
df20['municipio'].value_counts()

BRASILIA                6688
CURITIBA                4919
PORTO VELHO             4435
GUARULHOS               4223
BETIM                   3934
                        ... 
SALVADOR DAS MISSOES       1
ALCOBACA                   1
SERTAO                     1
URANDI                     1
VICENTINA                  1
Name: municipio, Length: 1789, dtype: int64

Em 2020 os municipios com mais acidentes foram, Brasilia com 6688 acidentes, Curitiba com 4919, Porto Velho com 4435, Guarulhos 4223 e Betim com 3934 acidentes. 

#### D)

In [22]:
df2019['feridos_graves'][(df2019['estado_fisico'] == 'Lesões Graves') & (df2019['uf'] == 'PB')].count()

1123

Na Paraíba tiveram o total de 1123 acidentes com lesões graves.

### <center> Tarefa 3 </center>

#### A)

Primeiro fiz a função dos estados de todos os anos.

In [23]:
estados = pd.concat([df2018[['uf']], df2019[['uf']], df20[['uf']]])

In [24]:
estados.value_counts()

uf
MG    146963
PR    118735
SC    101854
RS     74589
SP     68387
GO     58570
BA     55478
RJ     54203
MT     42913
PE     35702
RO     32903
MS     31569
ES     31041
CE     28329
PB     26449
MA     22820
PI     21952
PA     20254
RN     17763
DF     15746
TO     12824
AL      9050
SE      8845
RR      3936
AC      3708
AM      3459
AP      2915
dtype: int64

Depois, usei o metodo value_counts(), para saber a ocorrencia em cada estado e colocar em ordem 

In [43]:
estados.to_csv('estados.csv')

Tranformei o df de estados para csv.

#### B)

Primeiro vou fazer a junção dos 3 anos para um unico dataframe.

In [26]:
anos = pd.concat([df2018, df2019, df20], ignore_index = True)

Depois, fiz o ranking do dia da semana de cada ano.

In [27]:
df2018['dia_semana'].value_counts()

sábado           55378
domingo          54759
sexta-feira      49438
segunda-feira    42716
quinta-feira     42194
quarta-feira     40232
terça-feira      40092
Name: dia_semana, dtype: int64

In [28]:
df2019['dia_semana'].value_counts()

domingo          57096
sábado           55706
sexta-feira      51667
segunda-feira    43146
quinta-feira     42844
quarta-feira     40702
terça-feira      40505
Name: dia_semana, dtype: int64

In [29]:
df20['dia_semana'].value_counts()

domingo         68317
sábado          67187
sexta-feira     62075
segunda         51911
quinta-feira    49776
quarta-feira    48683
terça-feira     46533
Name: dia_semana, dtype: int64

E por fim, fiz o ranking dos 3 anos.

In [30]:
rankingtotal = anos['dia_semana'].value_counts()

In [31]:
rankingtotal

domingo          180172
sábado           178271
sexta-feira      163180
quinta-feira     134814
quarta-feira     129617
terça-feira      127130
segunda-feira     85862
segunda           51911
Name: dia_semana, dtype: int64

Transformei o rankingtotal em csv.

In [44]:
rankingtotal.to_csv('rankingtotal.csv')

### <center> Tarefa 4 </center>

#### A)

In [33]:
maisOcorrido = anos['causa_acidente']

In [34]:
maisOcorrido.value_counts()

Falta de Atenção à Condução                                       337949
Desobediência às normas de trânsito pelo condutor                 142169
Velocidade Incompatível                                           127360
Não guardar distância de segurança                                 81560
Ingestão de Álcool                                                 70466
                                                                   ...  
Pista em desnível                                                      3
Ingestão de álcool ou de substâncias psicoativas pelo pedestre         2
Problema com o freio                                                   2
Demais Fenômenos da natureza                                           2
Fumaça                                                                 2
Name: causa_acidente, Length: 67, dtype: int64

A causa mais frequente dos 3 anos juntos foi ***FALTA DE ATENÇÃO À CONDUÇÃO*** e as formas de acidente mais rara foram ***Ingestão de álcool ou de substâncias psicoativas pelo pedestre, Demais Fenômenos da natureza, Fumaça e Problema com o freio.***

#### B)

In [35]:
df2018[['ilesos', 'feridos_graves']].value_counts()


ilesos  feridos_graves
0       0                 165938
1       0                 125209
0       1                  33662
dtype: int64

In [36]:
len(df2018)

324809

A proporção de pessoas **ilesas** foi de:

In [37]:
125209/324809

0.38548500811245995

Com o value_counts() consegui o numero de feridos graves e os ilesos. Já com o len() consegui o numero total de acidentes.
A proporção de ilesos é de 0.38548500811245995 ou aproximadamente 38,54%

In [38]:
33662/324809

0.10363629086632452

A proporção de feridos graves é de 0.10363629086632452 ou aproximadamente 10,36%